### PR TITLE
Add Telemetry UI

### PR DESCRIPTION
**Pull request by romarickb**
## 📖 Description

This PR add [Telemetry UI](https://github.com/mirego/telemetry_ui) for : 
* HTTP requests
* GraphQL
* Absinthe
* Ecto
* System
 
Capture d’écran, le 2023-10-11 à 09 49 30

## 📝 Notes

For more information about Telemetry UI : https://github.com/mirego/telemetry_ui

## 🦀 Dispatch

- `#dispatch/elixir`

### CODE REVIEW

File: lib/elixir_boilerplate/telemetry_ui/telemetry_ui.ex [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
...586 2.7781986-6.268054 5.5391307-2.960203 4.141399-4.467906 6.623907-3.3519 14.833191.744003 5.472857 1.276531 10.50778 1.597582 15.104773l9.543032-27.726988c-.350835-1.412815-.642632-2.688766-.875391-3.8278548-.232758-1.1390887-.447848-2.446506-.645269-3.9222519z" fill="url(#g)"/><path d="m35.945755 13.009805c-2.422551 1.413992-4.299708 4.310913-5.631469 8.690763-1.331762 4.379852-2.550147 10.50277-3.655157 18.368756 1.47381-5.003069 2.451455-8.626771 2.932936-10.871105.722221-3.366501.968912-8.127131 2.886564-11.359156 1.278437-2.154684 2.434144-3.764436 3.467126-4.829258z" fill="#330a4c" fill-opacity=".316321"/><path d="m24.728788 59.937995c3.986659.569558 6.071303 1.075916 6.253931 1.519073.273942.664733-.504655 1.272785-2.717611.864177-1.475304-.272404-2.654077-1.06682-3.53632-2.38325z" fill="#fff"/><path d="m26.731652 5.3148253c-2.192807 2.6195801-4.092897 5.3967527-5.700271 8.3315187s-2.755892 5.124204-3.445555 6.568313c-.213753 1.077096-.318084 2.666371-.312993 4.767823.0051 2.101452.186856 4.438039.545295 7.009761.313817-5.035952 1.274508-9.924178 2.882072-14.664678 1.607565-4.740501 3.618049-8.7447464 6.031452-12.0127377z" fill="#ededed" fill-opacity=".603261"/></g></svg>
```
Comment (remi) [inline lib/elixir_boilerplate/telemetry_ui/telemetry_ui.ex:244]: Let’s use our own logo!

```suggestion
      <svg width="64" height="64" viewBox="0 0 266 266" xmlns="http://www.w3.org/2000/svg"><g fill-rule="nonzero" fill="none"><circle fill="#7F4BCB" cx="133" cy="133" r="133"/><path d="M135.08 60.197c-1.29-1.596-3.87-1.596-5.16 0C118.956 74.247 78 129.17 78 155.035 78 184.732 102.509 209 132.5 209s54.5-24.268 54.5-53.965c0-25.865-40.956-80.788-51.92-94.838Zm15.802 129.963c-.645.32-1.29.32-1.935.32-1.935 0-3.87-1.278-4.515-2.874-.968-2.555.322-5.11 2.58-6.387 19.026-7.663 17.414-30.654 17.092-30.974-.323-2.554 1.612-5.109 4.514-5.109 2.58-.32 5.16 1.597 5.16 4.47 1.29 10.219-2.58 32.252-22.896 40.554Z" fill="#3A186A"/></g></svg>
```

File: lib/elixir_boilerplate/telemetry_ui/telemetry_ui.ex [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+      """
+      # logo: """
+      #   <svg height="100" width="100"><circle cx="50" cy="50" r="40" stroke-width="3" fill="#8549a7" /></svg>
+      # """
```
Comment (remi) [inline lib/elixir_boilerplate/telemetry_ui/telemetry_ui.ex:248]:
```suggestion
```

**Review by simonprev — APPROVED** [review]
🔥 🔥 🔥 🔥

File: lib/elixir_boilerplate_web/router.ex [hunk 3]
```diff
@@ -15,6 +15,13 @@ defmodule ElixirBoilerplateWeb.Router do
     plug(:put_root_layout, {ElixirBoilerplateWeb.Layouts, :root})
   end
 
+  scope "/" do
+    pipe_through(:browser)
+    # Metrics can contains sensitive data you should protect it under authorization
+    # See https://github.com/mirego/telemetry_ui#security
+    get("/metrics", TelemetryUI.Web, [], assigns: %{telemetry_ui_allowed: true})
```
Comment (gcauchon) [inline lib/elixir_boilerplate_web/router.ex:22]: Should we `IO.warn/1` this message to the console so developer see an explicit message about securing the endpoint?

```suggestion
    # Metrics can contains sensitive data you should protect it under authorization
    # See https://github.com/mirego/telemetry_ui#security
    IO.warn("⚠️ TelemetryUI endpoint might need to be secured for production environnement!")
    get("/metrics", TelemetryUI.Web, [], assigns: %{telemetry_ui_allowed: true})
```

Comment (simonprev) [reply lib/elixir_boilerplate_web/router.ex:22]: That will break the linting task I think because our code warns on compile

Comment (gcauchon) [reply lib/elixir_boilerplate_web/router.ex:22]: Well we can either add a `credo` directive to disable the check, or add a section in the boilerplate `README` about security concerns to keep in mind before going to production.

Comment (remi) [reply lib/elixir_boilerplate_web/router.ex:22]: We could use `telemetry_ui_allowed: false` and add a comment to explicitely enable it wisely.

Comment (romarickb) [reply lib/elixir_boilerplate_web/router.ex]: Files are updated with your comments

File: README.md [hunk 4]
```diff
@@ -43,6 +44,20 @@ This boilerplate comes with batteries included, you’ll find:
...
+
+```elixir
+ # module ElixirBoilerplateWeb.Router
+ scope "/" do
+   pipe_through(:browser)
+   # Metrics can contains sensitive data you should protect it under authorization
+   get("/metrics", TelemetryUI.Web, [], assigns: %{telemetry_ui_allowed: true})
+ end
+```
+
```
Comment (remi) [inline README.md:60]: I don’t think we need this part in the "boilerplate README".

Comment (remi) [reply README.md:60]: We should actually document Telemetry UI here: https://github.com/mirego/elixir-boilerplate/blob/main/BOILERPLATE_README.md#-troubleshooting (and its french counterpart).

Comment (romarickb) [reply README.md:60]:
```
### Metrics

The project exposes a `GET /metrics` route. Dashboard display following metrics :
* HTTP requests
* Ecto
* GraphQL
* Absinthe 
```

Or better suggestion ?

Comment (remi) [reply README.md:60]: I would stay a bit more generic:

```md
### Metrics

The project exposes a [Telemetry UI](https://github.com/mirego/telemetry_ui) dashboard through the `GET /metrics` route that metrics configured [here](lib/elixir_boilerplate/telemetry_ui/telemetry_ui.ex).
```

**Pull request merged**